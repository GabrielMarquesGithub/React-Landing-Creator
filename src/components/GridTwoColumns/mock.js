export default {
  title: 'Grid two columns',
  text: 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Animi, illo minima. Nesciunt debitis et accusamus cupiditate, ipsam eaque, excepturi nobis deserunt inventore expedita quod beatae, voluptatibus eius quos autem natus.',
  image: 'assets/images/bgImg.svg',
  background: true,
};
