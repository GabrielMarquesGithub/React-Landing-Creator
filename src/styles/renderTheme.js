import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { theme } from './theme';

export const renderTheme = (element) => {
  return render(<ThemeProvider theme={theme}>{element}</ThemeProvider>);
};
