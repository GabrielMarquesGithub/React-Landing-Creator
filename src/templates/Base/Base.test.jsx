import { Base } from './index';
import { renderTheme } from '../../styles/renderTheme';

//mocks
import mock from '../../components/NavLinks/mock';
const logoData = {
  text: 'Alo galera de cowboy',
  link: 'https://www.google.com.br/',
  imageUrl: 'assets/images/logo.svg',
};

describe('<Base/>', () => {
  it('Should render Base component', () => {
    const { container } = renderTheme(
      <Base
        links={mock}
        logoData={logoData}
        footerHtml="Gabriel Marques MagalhÃ£es"
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
