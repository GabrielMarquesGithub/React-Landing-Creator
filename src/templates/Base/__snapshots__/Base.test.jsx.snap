// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Base/> Should render Base component 1`] = `
<div>
  <button
    aria-label="Open/Close menu"
    class="sc-ftvSup fdckYa"
  >
    <svg
      aria-hidden="true"
      aria-label="Open menu"
      class="StyledIconBase-ea9ulj-0 lbJwfL"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </button>
  <div
    class="sc-gKXOVf gMHglK"
  >
    <section
      class="sc-jSMfEi XnVFb"
    >
      <div
        class="sc-iBkjds iPRTCF"
      >
        <h1
          class="sc-bczRLJ eZrlHS"
        >
          <a
            class="sc-gsnTZi dTvfme"
            href="https://www.google.com.br/"
          >
            <img
              alt="Alo galera de cowboy"
              src="assets/images/logo.svg"
            />
          </a>
        </h1>
        <nav
          aria-label="Main menu"
          class="sc-eCYdqJ iCoeuS"
        >
          <div
            class="sc-hKMtZM ggUkCk"
          >
            <a
              class="sc-dkzDqf bVgcId"
              href="#target1"
              target="_self"
            >
              Link 1
            </a>
          </div>
          <div
            class="sc-hKMtZM ggUkCk"
          >
            <a
              class="sc-dkzDqf bVgcId"
              href="#target2"
              target="_self"
            >
              Link 2
            </a>
          </div>
          <div
            class="sc-hKMtZM ggUkCk"
          >
            <a
              class="sc-dkzDqf bVgcId"
              href="#target3"
              target="_self"
            >
              Link 3
            </a>
          </div>
          <div
            class="sc-hKMtZM ggUkCk"
          >
            <a
              class="sc-dkzDqf bVgcId"
              href="#target4"
              target="_self"
            >
              Link 4
            </a>
          </div>
          <div
            class="sc-hKMtZM ggUkCk"
          >
            <a
              class="sc-dkzDqf bVgcId"
              href="#target5"
              target="_self"
            >
              Link 5
            </a>
          </div>
          <div
            class="sc-hKMtZM ggUkCk"
          >
            <a
              class="sc-dkzDqf bVgcId"
              href="#target6"
              target="_self"
            >
              Link 6
            </a>
          </div>
          <div
            class="sc-hKMtZM ggUkCk"
          >
            <a
              class="sc-dkzDqf bVgcId"
              href="#target7"
              target="_self"
            >
              Link 7
            </a>
          </div>
          <div
            class="sc-hKMtZM ggUkCk"
          >
            <a
              class="sc-dkzDqf bVgcId"
              href="#target8"
              target="_self"
            >
              Link 8
            </a>
          </div>
        </nav>
      </div>
    </section>
  </div>
  <main
    class="sc-iqcoie crmuDU"
  >
    <footer
      class="sc-jqUVSM FZjPg"
    >
      <section
        class="sc-jSMfEi XnVFb"
      >
        <p
          class="sc-papXJ hSPKut"
        >
          Gabriel Marques Magalh√£es
        </p>
      </section>
    </footer>
  </main>
  <a
    aria-label="Go to top"
    class="sc-kDDrLX eBSPbA"
    href="#"
    title="Go to top"
  >
    <svg
      aria-hidden="true"
      class="StyledIconBase-ea9ulj-0 lbJwfL"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
      <path
        d="M7 14l5-5 5 5H7z"
      />
    </svg>
  </a>
</div>
`;
